fix(navigation): guard window.navigation for browsers without Navigation API (#11665)
remix-run/remix: packages/ui/.changes/patch.navigation-api-fallback.md

> Fall back to document navigation instead of crashing the client runtime in browsers that do not support the Navigation API (see #11665).

Reviewer: ```suggestion
Gracefully degrade to document navigations for browsers that do not support the Navigation API (see #11665)
```
